components/style/dom.rs

```diff
@@ -46,7 +46,23 @@ impl OpaqueNode {
 }
 
 pub trait TRestyleDamage : BitOr<Output=Self> + Copy {
-    fn compute(old: Option<&Arc<ComputedValues>>, new: &ComputedValues) -> Self;
+    /// The source for our current computed values in the cascade. This is a
+    /// ComputedValues in Servo and a StyleContext in Gecko.
+    ///
+    /// This must be obtained via the node.
+    type CascadeStyleSource;
```

Comment:
Also, please add a comment explaining why we do this dance with the style context (i.e. the inference from the caching).